File: app/src/main/java/org/cuieney/videolife/presenter/MusicHomePresenter.java
package org.cuieney.videolife.presenter;

import org.cuieney.videolife.common.api.WyApiService;
import org.cuieney.videolife.common.base.RxPresenter;
import org.cuieney.videolife.common.utils.RxUtil;
import org.cuieney.videolife.presenter.contract.MusicHomeContract;

import javax.inject.Inject;

import rx.Subscription;

/**
 * Created by cuieney on 17/3/4.
 */

public class MusicHomePresenter extends RxPresenter<MusicHomeContract.View> implements MusicHomeContract.Presenter {
    private WyApiService mRetrofitHelper;

    @Inject
    public MusicHomePresenter(WyApiService mRetrofitHelper) {
        this.mRetrofitHelper = mRetrofitHelper;
    }

    @Override
    public void getMusicData(String pagerNo) {
        Subscription rxSubscription = mRetrofitHelper.getMusicList(pagerNo)
                .compose(RxUtil.rxSchedulerHelper())
                .subscribe(musicListBean -> {
                    mView.showContent(musicListBean);
                }, throwable -> {
                    mView.error(throwable);
                });
        addSubscrebe(rxSubscription);
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/MusicHomePresenter.kt
package org.cuieney.videolife.kotlin.presenter

import org.cuieney.videolife.common.api.WyApiService
import org.cuieney.videolife.common.utils.RxUtil
import org.cuieney.videolife.entity.MusicListBean
import org.cuieney.videolife.kotlin.base.RxPresenter
import org.cuieney.videolife.kotlin.presenter.contract.MusicHomeContract
import javax.inject.Inject

/**
 * Created by cuieney on 2017/6/5.
 */
class MusicHomePresenter
@Inject constructor(val retrofitHelper: WyApiService) : RxPresenter(), MusicHomeContract.Presenter {

    override fun getMusicData(date: String, view: MusicHomeContract.View) {
        val rxSubscription = retrofitHelper.getMusicList(date)
                .compose<MutableList<MusicListBean>>(RxUtil.rxSchedulerHelper<MutableList<MusicListBean>>())
                .subscribe({ t -> view.showContent(t) }) { throwable -> view.error(throwable) }
        addSubscrebe(rxSubscription)
    }

}
